Feature - Custom Title

Zajozor opened the pull request:
Hi, thanks for the awesome GH action!

Not sure if you're a fan of a feature like this, but it would help us a lot - we track coverage for integration and unit tests separately and being able to configure this title would help us improve readability.

I'm also really sorry - I don't have a good way of testing this, but I'm fairly confident it does what it should.

ArtiomTr commented on src/report/getReportBody.ts:
```diff
@@ -84,7 +85,7 @@ export const getReportBody = (
             context.payload.pull_request?.head.sha ??
             context.sha,
-        dir: dir ? `for \`${dir}\`` : '',
+        name: customTitle || (dir ? `for \`${dir}\`` : ''),
```
With this code, the report header will look like this (assumed that `custom-title: Custom title`):

## Coverage report for Custom title

If this is the result you want to achieve, then the option should be named as `tag`, because it does not replace the entire title, only the tag in it. Also, don't forget to replace `dir` variable here:
https://github.com/ArtiomTr/jest-coverage-report-action/blob/19e9e06d96fefa23aef7ee979af01f60d89fb256/src/report/getReportBody.ts#L81
Because this is the place, where action identifies comment to update in on each re-run.

However, if you want the report title to look like this (assumed that `custom-title: Custom title`):

## Custom title

Then, you should replace this line:
https://github.com/ArtiomTr/jest-coverage-report-action/blob/19e9e06d96fefa23aef7ee979af01f60d89fb256/src/format/REPORT.md#L3
With the following:
```md
## {{ title }}
```
Then move the old title into the `strings.json` file and import it from `getReportBody.ts`. And then change the code like this:

```ts
    const reportBody = insertArgs(REPORT, {
        head: getReportTag(dir),
        //  changes here ↓
        title: insertArgs(title || customTitle, {
            dir: dir ? `for \`${dir}\`` : '',
        }),
        body: reportContent,
        sha:
            context.payload.after ??
            context.payload.pull_request?.head.sha ??
            context.sha,
        dir: dir ? `for \`${dir}\`` : '',
    });
```

I don’t know which solution will suit you better, I am satisfied with both options, so choose what you like best!

ArtiomTr reviewed (changes requested):
Thank you for your initiative! Your code looks great, but I have a couple of points to make.

First, if you would like to suggest a new feature, please create an issue first. We can discuss everything before we start implementing a new feature and thus save you time.

Secondly, don't forget to change the `action.yml` file. Add new arguments there, because without this, github won't allow you to specify additional parameters - it will just throw an error.

Last but not least, you can test your changes like this:

1. Create a repository, where you'll test your action. (or, you can fork [my repo, where I test new changes](https://github.com/ArtiomTr/jest-coverage-report-action-test))
2. Modify or create `.github/workflows/action.yml`:
```yml
name: 'coverage'
on:
    pull_request:
        branches:
            - master
jobs:
    coverage:
        runs-on: ubuntu-latest
        steps:
            - uses: actions/checkout@v1
#  your github username ↓                   your branch ↓            
            - uses: zajozor/jest-coverage-report-action@feat/custom-title
              with:
                  github_token: ${{ secrets.GITHUB_TOKEN }}
                  working_directory: project
                  threshold: 80
```
3. Create a PR, so your action will run. Don't care about commit naming, or branch naming, this repository will be your playground to test modifications to your code. **Also, don't forget, that this action does not automatically compile typescript, so you'll need to compile it manually, via running `npm run build`, and then commit `dist/index.js` file into your branch. Without that, action will not run your changes.**

I hope you take into account the possibility of testing your feature, as well as read my comment on your changes in the `getReportBody.ts` file, and choose the solution that suits you best.

Zajozor commented:
@ArtiomTr thanks for the wholesome reply, I really appreciate it :) !

I'll take a look at the comments, and implement the requested improvements in the upcoming days.

Zajozor commented:
@ArtiomTr I believe I addressed the changes and rebased above latest master :) I also tested it yesterday.

Please let me know if there is more to fix!

ArtiomTr reviewed (approved):
Everything is perfect 🚀

ArtiomTr commented:
@all-contributors please add @Zajozor for code
